app/assets/javascripts/discourse/app/components/d-editor.js

```diff
@@ -312,10 +311,24 @@ export default class DEditor extends Component {
             },
           };
 
-          this.menuInstance = this.menu.show(
-            virtualElementFromTextRange(),
-            menuOptions
-          );
+          const caretCoords =
+            this.textManipulation.autocompleteHandler.getCaretCoords(
+              this.textManipulation.autocompleteHandler.getCaretPosition()
+            );
+
+          const rect = document
+            .querySelector(".d-editor-input")
+            .getBoundingClientRect();
+
+          const virtualElement = {
+            getBoundingClientRect: () => ({
+              left: rect.left + caretCoords.left + 18,
+              top: rect.top + caretCoords.top + 10,
```

Comment:
I hate "magic" numbers, these are just some margins that work for both editors - open to suggestions

Reply:
Same I really hate them too. Sounds silly but maybe you could store them in consts directly above, like `const editorSafeLeftMargin = 18` and `const editorSafeTopMargin = 10`?